lib/pure/strmisc.nim

```diff
@@ -85,6 +86,76 @@ proc rpartition*(s: string, sep: string): (string, string, string)
 
   return partition(s, sep, right = true)
 
+
+func parseFloatThousandSep*(str: openArray[char]; sep = ','; decimalDot = '.'): float {.since: (1, 5).} =
+  ## Convenience func for `parseFloat` which allows for thousand separators,
+  ## this is designed to parse floats as found in the wild formatted for humans.
+  ##
+  ## The following assumptions and requirements must be met:
+  ## - `str` must not be empty.
+  ## - `str` must be stripped of trailing and leading whitespaces.
+  ## - `sep` must not be `'-'`.
+  ## - `decimalDot` must not be `'-'` nor `' '`.
+  ## - `sep` and `decimalDot` must be different.
+  ## - Exactly 1 separator must appear after each 3 consecutive inner digits to the left of the dot and nowhere else.
+  ##
+  ## See also:
+  ## * `parseFloat <strutils.html#parseFloat,string>`_
+  runnableExamples:
+    doAssert parseFloatThousandSep("-1,000") == -1000.0
+    doAssert parseFloatThousandSep("10,000.000") == 10000.0
+    doAssert parseFloatThousandSep("1,000,000.000") == 1000000.0
+    doAssert parseFloatThousandSep("10,000,000.000") == 10000000.0
+    doAssert parseFloatThousandSep("10.000,0", '.', ',') == 10000.0
+    doAssert parseFloatThousandSep("1'000'000,000", '\'', ',') == 1000000.0
+    doAssert parseFloatThousandSep("1000,000") == 1000000.0
+    doAssert parseFloatThousandSep("1000000") == 1000000.0
+  assert sep != '-' and decimalDot notin {'-', ' '} and sep != decimalDot
+
+  proc raiseError(i: int; c: char; s: openArray[char]) {.noinline, noreturn.} =
+    raise newException(ValueError,
+      "Invalid float containing thousand separators, invalid char $1 at index $2 for input $3" %
+      [c.repr, $i, s.repr])
+
+  if str.len > 1:
```

Comment:
why distinguish `if str.len > 1:` ? seems to work without this special case (so you can also remove `result = parseFloat($str[0])` below)

Reply:
To skip all the logic if its just `"0"` ?.

Reply:
ok fine; at least add a code comment "for performance" (otherwise seems like it's for semantics)

Reply:
EDIT: meh, I'm not sure singling this case out is really needed, but either way this leads to a bug in your code that needs fixing:
nim r --lib:lib --eval:'import strmisc; echo parseFloatThousandSep("")'

gives IndexDefect instead of ValueError
(because of $str[0])